docs/readthedocs/source/doc/PPML/Overview/ppml.md

```diff
@@ -222,39 +222,177 @@ The result should look like this:
 
 #### 2.2.3 Run Trusted Big Data and ML on Cluster
 
-##### 2.2.3.1 Configure the Environment
+WARNING: If you want spark standalone mode, please refer to [standalone/README.md](https://github.com/intel-analytics/BigDL/blob/main/ppml/trusted-big-data-ml/python/docker-graphene/standalone/README.md). But it is not recommended.
 
-Prerequisite: [ssh login without password](http://www.linuxproblem.org/art_9.html) for all the nodes.
+Follow the guide below to run Spark on Kubernetes manually. Alternatively, you can also use Helm to set everything up automatically. See [kubernetes/README.md](https://github.com/intel-analytics/BigDL/blob/main/ppml/trusted-big-data-ml/python/docker-graphene/kubernetes/README.md).
+
+##### 2.2.3.1 Prepare data, key and password
+
+Enter `BigDL/ppml/trusted-big-data-ml/python/docker-graphene` dir. Refer to the previous section about [preparing data, key and password](#2221-start-ppml-container). Then run the following commands to generate your enclave key and add it to your Kubernetes cluster as a secret. 
 
 ```bash
-nano environments.sh
+kubectl apply -f keys/keys.yaml
+kubectl apply -f password/password.yaml
+cd kubernetes
+bash enclave-key-to-secret.sh
 ```
-##### 2.2.3.2 Start Distributed Big Data and ML Platform
+##### 2.2.3.2 Prepare the k8s configurations
 
-Firstly run the following command to start the service:
+Create the RBAC:
 
 ```bash
-./deploy-distributed-standalone-spark.sh
+kubectl create serviceaccount spark
+kubectl create clusterrolebinding spark-role --clusterrole=edit --serviceaccount=default:spark --namespace=default
 ```
 
-Then run the following command to start the training:
+Generate k8s config file:
 
 ```bash
-./start-distributed-spark-train-sgx.sh
+kubectl config view --flatten --minify > /YOUR_DIR/kubeconfig
 ```
-##### 2.2.3.3  Stop Distributed Big Data and ML Platform
-
-Firstly stop the training:
+Create k8s secret, the secret created `YOUR_SECRET` should be the same as the password you specified in section 2.2.3.1:
 
 ```bash
-./stop-distributed-standalone-spark.sh
+kubectl create secret generic spark-secret --from-literal secret=YOUR_SECRET
 ```
 
-Then stop the service:
+##### 2.2.3.3  Start the client container
+
+Configure the environment variables in the following script before running it. Check [Bigdl ppml SGX related configurations](https://github.com/intel-analytics/BigDL/tree/main/ppml/trusted-big-data-ml/python/docker-graphene#1-bigdl-ppml-sgx-related-configurations) for detailed memory configurations. 
 
 ```bash
-./undeploy-distributed-standalone-spark.sh
-```
+export K8S_MASTER=k8s://$( sudo kubectl cluster-info | grep 'https.*' -o -m 1 )
+echo The k8s master is $K8S_MASTER .
+export ENCLAVE_KEY=/YOUR_DIR/enclave-key.pem

```

Comment:
remind user to modify your_xxxx

Reply:
update in https://github.com/intel-analytics/BigDL/pull/4515/files#diff-206dc464459fee64731e43f5909c3e6144c71a8f9711eec547deb6ccf7e404e7R246 and https://github.com/intel-analytics/BigDL/pull/4515/files#diff-206dc464459fee64731e43f5909c3e6144c71a8f9711eec547deb6ccf7e404e7R259